### PR TITLE
Basic Arabic phrases cheat sheet 

### PR DESCRIPTION
**Pull request by AhmedBenmoussa**
**What does your Instant Answer do?**
Shows some basic English phrases and their translation in Arabic. 

**What is the data source for your Instant Answer? (Provide a link if possible)**
[Linguanaut](http://www.linguanaut.com/english_arabic.htm)

**Why did you choose this data source?**
The link was mentioned on the original cheat sheet and when I saw that it has also an Arabic part I decided to use it. 

**Are there any other alternative (better) data sources?**
Yes, but not better than the data source that I used.  

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
For everyone who want to learn some basic phrases in Arabic or English.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
No. but it is a translation of a an existing cheat sheet (German cheat sheet). 

**Are you having any problems? Do you need our help with anything?**
No. Thank you.

**Where did you hear about DuckDuckHack? (For first time contributors)**
When I knew that DuckDuckGo is a opne source project.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
arabic

-----
IA Page: https://duck.co/ia/view/arabic_cheat_sheet

### CODE REVIEW

File: lib/DDG/Goodie/IsAwesome/AhmedBenmoussa.pm [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package DDG::Goodie::IsAwesome::AhmedBenmoussa;
```
Comment (moollaza) [inline lib/DDG/Goodie/IsAwesome/AhmedBenmoussa.pm]: Whoops! PR's should only add 1 new Instant Answer at a time. As well, we're no longer accepting PR's for the IsAwesome IA's. @AhmedBenmoussa could you please remove this Goodie from your PR?

**Comment by moollaza** [comment]
@AhmedBenmoussa thanks a lot for contributing, we really appreciate it! We'll be sure to review this shortly and give you some feedback!

We'll also have the community verify the translations to ensure everyone is in agreement :smiley:

**Comment by AhmedBenmoussa** [comment]
Hello @moollaza. First of all thank you for taking the time to answer me :) 
I want to ask you, if you don't mind, about the query: "arabic cheat sheet". What do you think. Is it appropriate or should I change it ? 

You want me to remove these two files **ahmedbenmoussa.pm** and **ahmedbenmoussa.t** ?

**Comment by moollaza** [comment]
>First of all thank you for taking the time to answer me :)

No problem, it's what I'm here for! Thanks for taking the time to contribute!

>I want to ask you, if you don't mind, about the query: "arabic cheat sheet". What do you think. Is it appropriate or should I change it ?

It's perfect. We used "german cheat sheet" for the German equivalent. I was thinking maybe we may need to make these more specific if we start converting to/from more languages, but for now it's OK.

>You want me to remove these two files ahmedbenmoussa.pm and ahmedbenmoussa.t ? 

Yes please.

**Comment by AhmedBenmoussa** [comment]
I've deleted them. 
Sorry @moollaza but can I ask you another question ?

**Comment by moollaza** [comment]
@AhmedBenmoussa sorry for the delay on this!

>Sorry @moollaza but I want to ask you another question ? Are you the developper of the "cheat sheet" >module ? 

Yes I wrote most of it, I'm a staff member at DuckDuckGo. What's your question?

**Comment by moollaza** [comment]
To give you an update:

I'll be putting the language cheat sheets onto beta.duckduckgo.com very soon and we'll then solicite feedback from the community on whether or not they approve of the translations. Once that's done and we've gathered some feedback, we'll need to implement any necessary changes and then this will be ready for merge.

Seeing as I'm not an Arabic speaker, we'll need to rely on the community to ensure everyone is happy with the translations here :+1:

**Comment by AhmedBenmoussa** [comment]
>Yes I wrote a good portion of it, I'm a staff member at DuckDuckGo. What's your question?

You answered my question. I just wanted to know if you're the developper of the "cheat sheet" goodie :) 
 
>I'll be putting the language cheat sheets onto beta.duckduckgo.com very soon and we'll then solicite feedback from the community on whether or not they approve of the translations. Once that's done and we've gathered some feedback, we'll need to implement any necessary changes and then this will be ready for merge.

I saw it on the DuckDuckGo beta platform :) 

>Seeing as I'm not an Arabic speaker, we'll need to rely on the community to ensure everyone is happy with the translations here

Yes of course :smiley: 
Thank you again Zaahir.

**Comment by saeidw** [comment]
Hello, I saw the cheat sheet on the beta site and noticed an error in the translation of
> Do you speak English?

Since all the translations are in Modern Standard Arabic, the correct translation should be
> هل تتكلم/تتكلمين اللغة الإنجليزية؟

The terminating ن on the feminine form of تتكلم is required.

**Comment by AhmedBenmoussa** [comment]
@saeidw Hello Saeid, 

Thank you for taking the time to check the cheat sheet :) I'm really grateful to you.
Yes you're right, the verb "تتكلم" on the feminine form must be completed by "ين" and not only a "ي". 

Like I translated the "Do you have a city map?" sentence: 
>هل تملك/تملكين خريطة للبلدة ؟   

I'll correct it.
Thank you again Saeid :smiley:

**Comment by iambibhas** [comment]
@moollaza I have a feedback on these language cheatsheets in general. When I search for, let's say, `arabic cheat sheet`, I'm kind of expecting to get a way to speak basic Arabic phrases without learning the whole language. And as I can't read Arabic, the cheatsheet in it's current form is pretty useless for me. Only people who can read Arabic can make use of this cheatsheet. But do they really need a cheatsheet for the basic phrases if they already know Arabic? 

For me, as a non-speaker, it'd be more useful if the phrases were also written with english characters showing how they are pronounced. I know it's not always possible to write a word from a language in english characters. I'm just expressing what I feel ddg should do if I am to search for this cheat sheet.

**Comment by seisfeld** [comment]
@iambibhas I see what you are saying. This is probably true for all cheat sheets that are not built with latin characters... i don't have an idea how to solve this though either...

**Comment by iambibhas** [comment]
> This is probably true for all cheat sheets that are not built with latin characters

Exactly. I was planning to add an cheatsheet for my native language Bengali. But kind of stuck with this confusion. Maybe we can introduce a new template type for language cheat sheets so that it can show an english phrase, exact translation in native characters and the pronunciation written in latin characters? Because I'd find that *immensely* helpful. We might need to organize them better so that they stay somewhat separated from other cheatsheets.

**Comment by AhmedBenmoussa** [comment]
@iambibhas You're right. I thought once to add the pronunciation of every sentence in English instead of the Arabic characters, but I wanted to respect the original cheat sheat (German cheat sheet).  I also tried to add the pronunciation with the Arabic translation, but the result was a little bit messy and not clean. 

These cheat sheets are for the persons who want to learn some basic English sentences (e.g., Arabic cheat sheet are for the Arabic people (and every person speaking Arabic) who want to learn some basic English sentences).

I'm working on another cheat sheet where you can see the Arabic alphabets and numbers with their pronunciation. I'm on the verification part, in case you want to learn the Arabic language :) 

>Maybe we can introduce a new template type for language cheat sheets so that it can show an english phrase, exact translation in native characters and the pronunciation written in latin characters

Yes, a template with a key and two values will be really great.

**Comment by moollaza** [comment]
@iambibhas I definitely agree. I'll speak to the team about this. Thanks for the suggestion :+1:

**Comment by moollaza** [comment]
@iambibhas the team agreed that allowing for English transliteration would be great. This is something that can be handled by the community in the open-source world, so if you'd like to implement a new template, or make an issue for the community to do so it would be great.

I think something as simple as putting the English letters bellow the translation would work:

arabic_cheat_sheet_at_duckduckgo

------

@AhmedBenmoussa it looks like this one is all done now? I'd be happy to merge if so?

File: share/goodie/cheat_sheets/arabic.json [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+   ],
+   "sections" : {
+      "Etiquette" : [
+         {
+            "val" : "تصبح/تصبحين على خير",
+            "key" : "Good Night"
+         },
+         {
+            "val" : "مساء الخير -من بعد الظهر حتى الغروب",
+            "key" : "Good Afternoon"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/arabic.json]: Let's lowercase "night", "afternoon" and "evening"

File: share/goodie/cheat_sheets/arabic.json [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+            "val" : "ِأرجوكَ/أرجوك",
+            "key" : "Please"
+         },
+         {
+            "val" : "! شكراً جزيلاً",
+            "key" : "Thank you very much!"
+         },
+         {
+            "val" : "صباح الخير",
+            "key" : "Good Morning"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/arabic.json]: Let's lowercase "morning" as well.

File: share/goodie/cheat_sheets/arabic.json [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+            "val" : "هل تتكلم/تتكلمين اللغة الإنجليزية ؟",
+            "key" : "Do you speak English?"
+         }, 
+         {
+            "val" : "هل يمكنكَ/يمكنكِ مساعدتي ؟",
+            "key" : "Can you help me?"
+         }, 
+         {
+            "val" : "! إتصل/إتصلي بالشرطة",
+            "key" : "Call the Police!"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/arabic.json]: "police" can be lowercased as well

**Comment by AhmedBenmoussa** [comment]
Hello @moollaza, 
I've made the changes that you asked me for. 

>it looks like this one is all done now? I'd be happy to merge if so?

I sent some screenshots of the cheat sheet to a person, asking him to check the sentences. Everything seems to be fine. So, yes, I think that it's ready to go :) 

Thank you again Zaahir.

**Comment by moollaza** [comment]
@AhmedBenmoussa excellent!

**Comment by moollaza** [comment]
@AhmedBenmoussa congrats! It's live: https://duckduckgo.com/?q=arabic+cheat+sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Comment by AhmedBenmoussa** [comment]
@moollaza Sorry for my late answer. I was busy these days. 
I will post other cheat sheets when I'll find the time. I will spare no effort to help the DDG community.  

Just a question Zaahir, what about the language template. is it up ? can we use it ?  
Thank you again for your help :)

**Comment by moollaza** [comment]
> I will post other cheat sheets when I'll find the time. I will spare no effort to help the DDG community. 

Thank you very much!

> Just a question Zaahir, what about the language template. is it up ? can we use it ?

Yes! It has been merged and the other languages have been moved into a new subdir. You can definitely use the new template!

**Pull request merged**